src/apps/service/apps/de/jobs.clj

```diff
@@ -51,6 +40,13 @@
              :parent_id          (:parent_id submission))
       (jp/save-job submission)))
 
+(defn- job-type-for-step
+  "Determines the job type to use for a single job step."
+  [{{tool-type :type} :component}]
+  (if (= tool-type c/interactive-tool-type)
+    jp/interactive-job-type
+    jp/de-job-type))
```

Comment:
For de-only apps, we no longer have access to the app information by the time we get here. The choice was either to ensure that the app information was available here or to make the determination again based on the tool type. I chose the latter for the time being.